addons/event_sale/models/event_registration.py

```diff
@@ -97,7 +98,7 @@ def write(self, vals):
     def _synchronize_so_line_values(self, so_line):
         if so_line:
             return {
-                'partner_id': so_line.order_id.partner_id.id,
+                'partner_id': False if request and request.env.user._is_public() else so_line.order_id.partner_id.id,
```

Comment:
Have you tried `self.env.user._is_public()` ?

Since we are outside of a controller, I would rather not use request.

Reply:
i used this for note can be logged by odoobot but now we decide the note can be logged by Public user so i remove this change. Thanks